### PR TITLE
Show batch file path in error messages for sqlcmd DDL batches

### PR DESCRIPTION
**Pull request by wolffcm**
I had to do some minor refactoring of Xin's recent commit to make this work.

### CODE REVIEW

File: src/frontend/org/voltdb/compiler/VoltCompiler.java [hunk 1]
```diff
@@ -1101,10 +1102,16 @@ private void compileDatabase(
...
+            String origFilename = m_currentFilename;
+            if (m_currentFilename.equals(NO_FILENAME))
+                m_currentFilename = schemaReader.getName();
+
             // add the file object's path to the list of files for the jar
             m_ddlFilePaths.put(schemaReader.getName(), schemaReader.getPath());
 
             ddlcompiler.loadSchema(schemaReader, db, whichProcs);
+
+            m_currentFilename = origFilename;
```
Comment (paulmartel) [inline src/frontend/org/voltdb/compiler/VoltCompiler.java]: Should the m_currentFilename restore be in a finally block?

File: tests/sqlcmd/.gitignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/scripts/*/*.err
+/scripts/*/*.errclean
+/scripts/*/*.out
+/scripts/*/*.outclean
+/sqlcmdtest.report
```
Comment (paulmartel) [inline tests/sqlcmd/.gitignore]: Technically, the scripts directory could be of arbitrary depth, so scripts/*/ would not cover them -- I suppose we could just extend the pattern later as needed.  OTOH, I don't know that anyone would object to these extensions being universally ignored. What about .errdiff and .outdiff files?

File: tests/sqlcmd/scripts/batching/redundant_table_batch.in [hunk 3]
```diff
@@ -0,0 +1 @@
+file -batch ./scripts/batching/redundant_table.sqlfile
```
Comment (paulmartel) [inline tests/sqlcmd/scripts/batching/redundant_table_batch.in]: Unless a .sqlfile is used in multiple test cases, I would have named it exactly like its .in file, so like
file_batch_positive.sqlfile and redundant_table_batch.sqlfile

**Comment by paulmartel** [comment]
End of review round -- no show-stoppers.

**Comment by wolffcm** [comment]
I pushed another commit since Jenkins found an NPE in my change to VoltCompiler.

File: tests/sqlcmd/.gitignore (hunk 2 above)
Comment (wolffcm) [inline tests/sqlcmd/.gitignore]: I guess I'm just being conservative.  In general I want git to remind me if there's something that I maybe want to add to source control, so it seems dangerous to have an overly broad .gitignore.

I'm fine with adding patterns to ignore outdiff and errdiff files too.

File: tests/sqlcmd/scripts/batching/redundant_table_batch.in (hunk 3 above)
Comment (wolffcm) [inline tests/sqlcmd/scripts/batching/redundant_table_batch.in]: Good idea.  I'll change the name to match when I push commits for my next ticket.

**Comment by wolffcm** [comment]
Are you okay with me merging this?  My branch for inline batching depends on these commits because there was significant sqlcmd refactoring here for batches.

**Pull request merged**